lib/ansible/modules/utilities/logic/wait_for.py

```diff
@@ -113,63 +99,66 @@
       - This overrides the normal error message from a failure to meet the required conditions.
 notes:
   - The ability to use search_regex with a port connection was added in 1.7.
-requirements: []
 author:
-    - "Jeroen Hoekx (@jhoekx)"
-    - "John Jarvis (@jarv)"
-    - "Andrii Radyk (@AnderEnder)"
+    - Jeroen Hoekx (@jhoekx)
+    - John Jarvis (@jarv)
+    - Andrii Radyk (@AnderEnder)
 '''
 
-EXAMPLES = '''
+EXAMPLES = r'''
 
-# wait 300 seconds for port 8000 to become open on the host, don't start checking for 10 seconds
-- wait_for:
+- name: Wait 300 seconds for port 8000 to become open on the host, don't start checking for 10 seconds
+  wait_for:
     port: 8000
     delay: 10
 
-# wait 300 seconds for port 8000 of any IP to close active connections, don't start checking for 10 seconds
-- wait_for:
+- name: Wait 300 seconds for port 8000 of any IP to close active connections, don't start checking for 10 seconds
+  wait_for:
     host: 0.0.0.0
     port: 8000
     delay: 10
     state: drained
 
-# wait 300 seconds for port 8000 of any IP to close active connections, ignoring connections for specified hosts
-- wait_for:
+- name: Wait 300 seconds for port 8000 of any IP to close active connections, ignoring connections for specified hosts
+  wait_for:
     host: 0.0.0.0
     port: 8000
     state: drained
     exclude_hosts: 10.2.1.2,10.2.1.3
 
-# wait until the file /tmp/foo is present before continuing
-- wait_for:
+- name: Wait until the file /tmp/foo is present before continuing
+  wait_for:
     path: /tmp/foo
 
-# wait until the string "completed" is in the file /tmp/foo before continuing
-- wait_for:
+- name: Wait until the string "completed" is in the file /tmp/foo before continuing
+  wait_for:
     path: /tmp/foo
     search_regex: completed
 
-# wait until the lock file is removed
-- wait_for:
+- name: Wait until the lock file is removed
+  wait_for:
     path: /var/lock/file.lock
     state: absent
 
-# wait until the process is finished and pid was destroyed
-- wait_for:
+- name: Wait until the process is finished and pid was destroyed
+  wait_for:
     path: /proc/3466/status
     state: absent
 
-# output customized message when failed
-- wait_for:
+- name: Output customized message when failed
+  wait_for:
     path: /tmp/foo
     state: present
     msg: Timeout to find file /tmp/foo
 
-# wait 300 seconds for port 22 to become open and contain "OpenSSH",
-# don't assume the inventory_hostname is resolvable and don't start checking for 10 seconds
-- wait_for: port=22 host="{{ ansible_host|default(ansible_ssh_host|default(inventory_hostname)) }}" search_regex=OpenSSH delay=10
-  connection: local
+# Don't assume the inventory_hostname is resolvable and delay 10 seconds at start
+- name: Wait 300 seconds for port 22 to become open and contain "OpenSSH"
+  wait_for:
+    port: 22
+    host: '{{ ansible_ssh_host | default(inventory_hostname) }}'
+    search_regex: OpenSSH
+    delay: 10
+  delegate_to: localhost
```

Comment:
Correct. But this will fail in a heterogeneous environment (mixing Linux and Windows).
So `local_action` is actually preferred.

You see what a mess this is :-/